docs/general-usage/middleware.rst

```diff
@@ -0,0 +1,71 @@
+Middleware
+==========
+
+You can use ``middleware`` to affect the evaluation of fields in your schema.
+
+A middleware is any object or function that responds to ``resolve(next_middleware, *args).``
+
+Inside that method, it should either:
+
+    - Send ``resolve`` to the next middleware to continue the evaluation; or
+    - Return a value to end the evaluation early.
+
+
+Resolve arguments
+-----------------
+
+Middlewares ``resolve`` is invoked with several arguments:
+
+- ``next`` represents the execution chain. Call ``next`` to continue evaluation.
+- ``root`` is the root value object passed throughout the query.
+- ``info`` is the resolver info.
+- ``args`` is the dict of arguments passed to the field.
+
+
+GrappleMiddleware
+-----------------
+.. module:: grapple.middleware
+.. class:: GrappleMiddleware(object)
+
+::
```

Comment:
nitpick: I prefer the `.. code-block:: python` notation as we get nicer syntax highlighting

note that I didn't raise it for the settings docs as they were small enough, but I think little details like this make for a nicer reading experience.

Also it is worth adding a `# settings.py` line before `GRAPHENE = {` to softly indicate this goes into settings, unless you spell it out